### ISSUE TITLE
auditwheel stopped recognizing platform tags

### ISSUE THREAD

**Issue by nightlark** [pypa/auditwheel#247]
We're using the manylinux2010 docker image, and for our latest release, auditwheel stopped working.

tl;dr auditwheel 3.1.1 broke recognizing platform wheels that previously worked fine.

The platform tags back then looked like helics-2.5.0-cp37-cp37m-linux_x86_64.whl and running auditwheel worked fine. Now builds are failing with this error:
```
INFO:auditwheel.main_repair:Repairing helics-2.5.1-cp37-cp37m-linux_x86_64.whl
INFO:auditwheel.main_repair:This does not look like a platform wheel
```

The exact command run is: `auditwheel repair "$whl" --plat manylinux2010_x86_64 -w wheelhouse/manylinux2010`

I tested with some older manylinux2010 images, the last one with an auditwheel that works is `pypa/manylinux2010_x86_64:2020-04-25-bdd9229`. Everything from `pypa/manylinux2010_x86_64:2020-04-29-0e1afc5` onward has an auditwheel that doesn't work.

**Comment by lkollar** [comment]
Can you upload `helics-2.5.1-cp37-cp37m-linux_x86_64.whl` somewhere so that I can reproduce this?

**Comment by nightlark** [comment]
I'll see if I can get one off our CI system. I looked at the changes made in the auditwheel 3.1.1 release and it wasn't obvious that one of them would cause the problem, so checking out the artifacts could help confirm what happened.

**Comment by nightlark** [comment]
So, good news! After getting a copy of the wheel I found the actual cause of the error.

There was a change elsewhere that made the compiled Python extension end up in the wrong place so it wasn't even getting put inside the wheel (it just happened that 3.1.1 detects this scenario and has a non-zero exitcode that failed the CI builds, whereas running with an old Docker image had an auditwheel that made it look like there was no error).

**Comment by moble** [comment]
Ah, thank you!  That seems obvious in retrospect.  I have a similar problem, except my binary doesn't appear anywhere on the system...
